server.go

```diff
@@ -1669,6 +1694,29 @@ func (s *Server) loadClients(v []storage.Client) {
 	}
 }
 
+// newClientFromStorage creates a Client from a storage.Client.
+func (s *Server) newClientFromStorage(c *storage.Client) *Client {
```

Comment:
I'm not against it, but what's the intention behind extracting this into an additional method if it's only called from the same place?